src/sqlfluff/rules/L006.py

```diff
@@ -63,8 +63,9 @@ def _missing_whitespace(seg, before=True):
         if seg.is_whitespace:
             return False
         # And it's not an opening/closing bracket
-        if seg.name.endswith("_bracket"):
-            if seg.name.startswith("start_" if before else "end_"):
+        seg_type = seg.get_type()
```

Comment:
Is there a reason for requiring a call to `.get_type()` rather than exposing the type directly as a property? (Note that Python "properties" can actually be set up to call a function.)

https://www.programiz.com/python-programming/property

If the intent is to remind people that types are a bit complex (e.g. that sometimes you need to use `.is_type(blah)` rather than `.type == blah`), that totally makes sense. 👍

Reply:
Yeah I'm thinking the latter. We're moving to a world where a segment doesn't just have _one_ type - it has a specific one _and_ and all of the types of it's parent segments.

If we just check `.type` then we only get the last one (and in the case of `KeywordSegment`, not even that) - so by consistently called `.get_type()` we always get the effect we want.